systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/ServiceUtils.java

```diff
@@ -76,12 +77,24 @@ public static void waitForNodePortService(String serviceName) throws Interrupted
      * Wait until Service of the given name will be deleted.
      * @param serviceName service name
      */
-    public static void waitForServiceDeletion(String serviceName) {
-        LOGGER.info("Waiting for Service {} deletion in namespace {}", serviceName, kubeClient().getNamespace());
+    public static void waitForServiceDeletion(String serviceName, String namespace) {
+        LOGGER.info("Waiting for Service {} deletion in namespace {}", serviceName, namespace);
 
         TestUtils.waitFor("Service " + serviceName + " to be deleted", Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, DELETION_TIMEOUT,
-            () -> kubeClient().getService(serviceName) == null);
-        LOGGER.info("Service {} in namespace {} was deleted", serviceName, kubeClient().getNamespace());
+            () -> kubeClient().namespace(namespace).getService(serviceName) == null);
+        LOGGER.info("Service {} in namespace {} was deleted", serviceName, namespace);
+    }
+
+    /**
+     * Wait until Service of the given name will be created.
+     * @param serviceName service name
+     */
+    public static void waitForServiceCreation(String serviceName, String namespace) {
```

Comment:
Do we need to wait for service creation or we can just simply wait for some Kafka readiness with clients?

Reply:
I think this will be removed, as this whole part of service killing is redundant (as per Jakub's comment above) and it just creates problems sometimes - I was unable to debug the issue.